### DISCUSSION TITLE
Plot of radiation evolution in time

### DISCUSSION THREAD

**Discussion by kaklik** [Open-Gamma-Project/Gamma-MCA#48] [Ideas]
For some experiments set up, seeing a time evolution of radiation flux is valuable.  For example, it could allow finding the best setup to get the most cps from the specific isotope or simply monitoring the time evolution of radiation in real time. 

The graph could be implemented as the sum of cts of specified channels or energy ranges. Then the sum should be plotted in real-time in a running graph. 
The channel range should be selectable similarly to the actual energy range selection of view in the current spectrogram.  There is also helpful if there could be possible to select multiple channel ranges and plot their sum by multiple plot lines.

**Comment by NuclearPhoenixx** [comment]
Plotting the time evolution of the total cps is pretty easy as there already is a cpsValues array, which stores the activity per plot refresh in a chronological order.

Selecting ranges for that is a much more difficult task and will need for me to complete the ROI stats first as it uses data selections too. Then it would be possible to modify that to do some kind of activity evolution chart.

**Comment by NuclearPhoenixx** [comment]
A simple evolution chart has been added with version 2023-03-04. You can have a look at it in the "Serial Connection" tab. Have fun!

**Reply by kaklik** [reply]
Yes, that is an improvement of the GammaMCA. We merged it and tried to use it with our instrument which does not publish the data every second. 
Currently, the resulting time graph looks like the following:

![LABDOS01 time graph](https://user-images.githubusercontent.com/1194177/226102335-08abb63c-b9a7-4e21-83cf-b9f04fc1f521.png)

As can be seen, the moving average does count with "zeros" between the received data. 
Is there some mechanism in GammaMCA that needs to plot the data every second?

**Reply by NuclearPhoenixx** [reply]
Shouldn't be a problem if you change the "Plot Refresh Time" in the settings to a value more suited to your needs.

**Comment by roman-dvorak** [comment]
Hi, this activity evolution graph is very interesting. Thank you for addition. 

I have a few notes about that:

- the graph shows a new point every second. I don't know what the reason is. It would be better for me to show new point in the time of recieving new packet/spectrum from detector. Value can be divided by time since last spectrum. To preserve CPS units. 

- It would be more appropriate to display only points without a lines in graph. 

Thank you. Those are my initial notes from user point of view.

**Reply by NuclearPhoenixx** [reply]
It would be entirely possible to only show points in the evolution graph that are not equal to zero. However, that would entirely eliminate the possibility of checking for device problems or anything where 0 can be a "valid" outcome. As it is now, the idea is that you set the correct "Plot Refresh Time" in the settings to some sensible value for you. If you expect data to come in every 20 secs or so, just set it to 30 and you're fine. Doesn't make too much sense to refresh it every second if you only get data sporadically.

As for the lines, I'm open for suggestions. If more people think it would be better without them, I'll remove them. As easy as that.

**Comment by NuclearPhoenixx** [comment]
I'm closing this for now since there haven't been any new comments. If anyone has any opinions or suggestions for this feature, feel free to post and I'll reopen this topic.
